cdk_framework/EKS/lib/utils/validate.ts

```diff
@@ -0,0 +1,147 @@
+const yaml = require('js-yaml')
+
+const supportedFields = new Set(['version', 'ec2_instance', 'launch_type', 'node_size'])
+const supportedVersions = new Set(['1.18', '1.19', '1.20', '1.21']);
+const supportedCPUArchitectures = new Set(['m5', 'm6g', 't4g']);
+const supportedLaunchTypes = new Set(['ec2', 'fargate']);
+const supportedNodeSizes = new Set(['medium', 'large', 'xlarge', '2xlarge', '4xlarge', '8xlarge', '12xlarge', '16xlarge', '24xlarge', 'metal']);
+const supportedT4gInstances = new Set(['nano', 'micro', 'small', 'medium', 'large', 'xlarge', '2xlarge'])
+const requiredFields = new Set(['version', 'launch_type'])
+
+
+export function validateClustersConfig(info: Object){
+    const data = Object(info)
+    if(!data['clusters']){
+        throw new Error('No clusters field being filed in the yaml file')
+    }
+    const clusterInfo = data['clusters']
+    const clusterNamesSet = new Set()
+    for(const [key, value] of Object.entries(clusterInfo)){
+        
+        const val = Object(value)
+        if(clusterNamesSet.has(key)){
+            throw new Error('Cannot have multiple clusters with the samne name')
+        }
+        clusterNamesSet.add(key)
+        validateRequiredFields(val)
+        for(const [k, v] of Object.entries(val)){
+            if(!supportedFields.has(k)){
+                throw new Error("Uncompatible field type")
+            }
+            switch(k){
+                case 'version':
+                    val[k] = validateVersion(String(v))
+                    break;
+                case 'launch_type':
+                    val[k] = convertAndValidateLaunchType(val)
+                    break
+            }
+        }
+    }
+}
+
+function validateRequiredFields(fields: any){
+    for(const field of requiredFields){
+        if(!fields[field]){
+            throw new Error('Required field - ' + field + ' - not provided')
+        }
+    }
+}
+
+function checkToSetUpDefaults(fields: any){
+    if(!fields['ec2_instance']){
+        fields['ec2_instance'] = 'm5'
+    }
+    if(!fields['node_size']){
+        fields['node_size'] = 'large'
+    }
+}
+
+function validateVersion(version: string){
+    
+    if(version === '1.2'){
+        version = '1.20'
+    }
+
+    if(!supportedVersions.has(version)){
+        throw new Error('Version needs to be number between 1.18 to 1.21');
```

Comment:
```suggestion
        throw new Error('Version needs to be number between 1.18 to 1.21');
```
Can we convert our supported version set to string instead and print it out?